freqtrade/data/dataprovider.py

```diff
@@ -483,6 +502,44 @@ def ohlcv(
         else:
             return DataFrame()
 
+    def trades(
+        self,
+        pair: str,
+        timeframe: Optional[str] = None,
+        copy: bool = True,
+        candle_type: str = ''
+    ) -> DataFrame:
+        """
+        Get candle (TRADES) data for the given pair as DataFrame
+        Please use the `available_pairs` method to verify which pairs are currently cached.
+        :param pair: pair to get the data for
+        :param timeframe: Timeframe to get data for
+        :param candle_type: '', mark, index, premiumIndex, or funding_rate
+        :param copy: copy dataframe before returning if True.
+                     Use False only for read-only operations (where the dataframe is not modified)
+        """
+        if self._exchange is None:
+            raise OperationalException(NO_EXCHANGE_EXCEPTION)
+        if self.runmode in (RunMode.DRY_RUN, RunMode.LIVE):
+            _candle_type = CandleType.from_string(
+                candle_type) if candle_type != '' else self._config['candle_type_def']
+            return self._exchange.trades(
+                (pair, timeframe or self._config['timeframe'], _candle_type),
+                copy=copy
+            )
+        elif self.runmode in (RunMode.BACKTEST, RunMode.HYPEROPT):
+            _candle_type = CandleType.from_string(
+                candle_type) if candle_type != '' else self._config['candle_type_def']
+            data_handler = get_datahandler(
+                self._config['datadir'], data_format=self._config['dataformat_trades'])
+            ticks = data_handler.trades_load(pair)
+            trades_df = public_trades_to_dataframe(
+                ticks.values.tolist(), pair=pair)
+            return trades_df
```

Comment:
It's a leftover mainly, but I left it in there because `public_trades_to_dataframe` also converts columns to right types e.g.
``` python
# Some exchanges return int values for Volume and even for OHLC.
    # Convert them since TA-LIB indicators used in the strategy assume floats
    # and fail with exception...
    df = df.astype(dtype={'amount': 'float', 'cost': 'float', 'price': 'float'})
```

Reply:
i don't think types out of `data_handler.trades_load()` are wrong ...